src/libaktualizr/uptane/tuf.cc

```diff
@@ -361,13 +361,19 @@ void Uptane::Snapshot::init(const Json::Value &json) {
         it.key().asString().substr(0, it.key().asString().rfind('.'));  // strip extension from the role name
     auto role_object = Role(role_name, !Role::IsReserved(role_name));
 
-    role_size_[role_object] = meta_size.asInt64();
     if (meta_version.isIntegral()) {
       role_version_[role_object] = meta_version.asInt();
     } else {
       role_version_[role_object] = -1;
     }
 
+    // Size and hashes are not required, but we may as well record them if
```

Comment:
Can we assume that hashes_list and meta_size won't return true for isObject() if there is no "hashes" or "length" in the json/meta_list ?

Reply:
Yep, `isObject` is false if the field is not present.